### ISSUE TITLE
Debian 9 The current .NET SDK does not support targeting .NET Core 2.1

### ISSUE THREAD

**Issue by charles-strong** [dotnet/cli#10227]
## Steps to reproduce
On Debian 9
follow the install steps
[Install .NET Core 2.1 SDK on Linux Debian 9 x64](https://www.microsoft.com/net/download/linux-package-manager/debian9/sdk-2.1.403)
and here
[Install .NET Core 2.1 Runtime on Linux Debian 9 x64](https://www.microsoft.com/net/download/linux-package-manager/debian9/runtime-2.1.5)

gathered from this page
[.NET Core 2.1 downloads](https://www.microsoft.com/net/download/dotnet-core/2.1)

try running `dotnet restore` against this simple .csproj

```
<Project Sdk="Microsoft.NET.Sdk.Web">
<PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
</Project>
```

## Expected  behavior

restore should complete without error
## Actual behavior
`/usr/share/dotnet/sdk/2.1.4/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.TargetFrameworkInference.targets(135,5): error : The current .NET SDK does not support targeting .NET Core 2.1.  Either target .NET Core 2.0 or lower, or use a version of the .NET SDK that supports .NET Core 2.1. `

also tried with SDK V2.1.3 from this response found on a other SO question.

> I came across this post The current .NET SDK does not support targeting .NET Core 2.1 on developercommunity.visualstudio.com where a Microsoft support person explains the confusion:

> "Thank you for your feedback! We have determined that this issue is not a bug. The first SDK with .NET Core 2.1 support is 2.1.300-preview1. We know the versioning is confusing which is why starting in 2.1.300, the major.minor versions of the SDK will now be aligned with the major.minor versions of the runtime as well."

[The current .NET SDK does not support targeting .NET Core 2.1. target .NET Core 2.0 or lower or use a vs of the .NET SDK that support NET Core 2.1](https://stackoverflow.com/questions/49171623/the-current-net-sdk-does-not-support-targeting-net-core-2-1-target-net-core)

`/dotnet/sdk/2.1.3/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.TargetFrameworkInference.targets(135,5): error : The current .NET SDK does not support targeting .NET Core 2.1.  Either target .NET Core 2.0 or lower, or use a version of the .NET SDK that supports .NET Core 2.1. `

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.1.4)

Product Information:
 Version:            2.1.4
 Commit SHA-1 hash:  5e8add2190

Runtime Environment:
 OS Name:     debian
 OS Version:  9
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.4/

Host (useful for support):
  Version: 2.1.5
  Commit:  290303f510

.NET Core SDKs installed:
  2.1.3 [/usr/share/dotnet/sdk]
  2.1.4 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by charles-strong** [comment]
Roll back to a RC version is a work-around for this this issue. `sudo apt-get install dotnet-sdk-2.1.300-rc1-008673 `

**Comment by livarcocc** [comment]
2.1.300 > 2.1.4. We treat the patch version as a full number.

For instance, our latest released SDK is 2.1.403.
